manifests/generate.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+# This script updates the manifsts in this directory using helm.
+# Values files for the manifests in this directory can be found in 
+# ../calico/charts/values.
+
+HELM=${HELM:-../bin/helm}
+
+##########################################################################
+# Build the operator manifest. 
+##########################################################################
+cat <<EOF > tigera-operator.yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: tigera-operator
+  labels:
+    name: tigera-operator
+EOF
+
+${HELM} -n tigera-operator template \
+	--include-crds \
+	../charts/tigera-operator \
+	-f ../charts/values/tigera-operator.yaml >> tigera-operator.yaml
+
+##########################################################################
+# Build CRD manifest.
+#
+# This manifest is used in "Calico the hard way" documentation.
+##########################################################################
+echo "# CustomResourceDefinitions for Calico the Hard Way" > crds.yaml
+for FILE in $(ls ../charts/calico/crds); do
+	${HELM} template ../charts/calico \
+		--include-crds \
+		--show-only $FILE \
+		-f ../charts/values/calico.yaml \
+		-f ../charts/values/values.common.yaml >> crds.yaml
+done
+
+##########################################################################
+# Build Calico manifests.
+#
+# To add a new manifest to this directory, define
+# a values file in ../charts/values/ and then add it to VALUES_FILES.
+##########################################################################
+VALUES_FILES="calico-typha.yaml
+	calico-bpf.yaml
+	calico-vxlan.yaml
+	calico.yaml
+	calico-etcd.yaml 
+	calico-policy-only.yaml
+	flannel-migration/calico.yaml
+	"
```

Comment:
It'd be neat if we iterated the files in ../charts/values automatically instead of needing to also update this list. We could even use the values filename as the output filename.

Reply:
I ended up doing this